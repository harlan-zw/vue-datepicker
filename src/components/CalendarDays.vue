<script setup lang="ts">
import { ref } from "vue";
import type { Ref } from "vue";

import type { HeaderDay } from "~/types";

const days: Ref<HeaderDay[]> = ref([
  { key: 1, name: "Mo" },
  { key: 2, name: "Tu" },
  { key: 3, name: "We" },
  { key: 4, name: "Th" },
  { key: 5, name: "Fr" },
  { key: 6, name: "Sa" },
  { key: 0, name: "Su" },
]);
</script>

<template>
  <ul class="calendar_wrapper_content-header-days">
    <li v-for="day in days" :key="day.key" class="font-medium">
      {{ day.name }}
    </li>
  </ul>
</template>

<style>
.calendar_wrapper_content-header-days {
  @apply grid grid-cols-7 text-center py-6 text-sm;
}
</style>
