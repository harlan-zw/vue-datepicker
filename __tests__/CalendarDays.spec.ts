import { afterEach, beforeEach, describe, expect, it } from "vitest";

import { mount } from "@vue/test-utils";
import CalendarDays from "~/components/CalendarDays.vue";

let wrapper: any;

beforeEach(() => {
  wrapper = mount(CalendarDays);
});

afterEach(() => {
  wrapper.unmount();
});

describe("CalendarDays", () => {
  it("is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("renders CalendarDays properly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
