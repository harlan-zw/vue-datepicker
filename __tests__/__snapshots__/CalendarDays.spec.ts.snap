// Vitest Snapshot v1

exports[`CalendarDays > must be a calendar-days list 1`] = `
"<ul class=\\"calendar_wrapper_content-header-days\\">
  <li class=\\"font-medium\\">Mo</li>
  <li class=\\"font-medium\\">Tu</li>
  <li class=\\"font-medium\\">We</li>
  <li class=\\"font-medium\\">Th</li>
  <li class=\\"font-medium\\">Fr</li>
  <li class=\\"font-medium\\">Sa</li>
  <li class=\\"font-medium\\">Su</li>
</ul>"
`;

exports[`CalendarDays > must be a svg 1`] = `
"<ul class=\\"calendar_wrapper_content-header-days\\" name=\\"chat\\">
  <li class=\\"font-medium\\">Mo</li>
  <li class=\\"font-medium\\">Tu</li>
  <li class=\\"font-medium\\">We</li>
  <li class=\\"font-medium\\">Th</li>
  <li class=\\"font-medium\\">Fr</li>
  <li class=\\"font-medium\\">Sa</li>
  <li class=\\"font-medium\\">Su</li>
</ul>"
`;

exports[`CalendarDays > renders CalendarDays properly 1`] = `
"<ul class=\\"calendar_wrapper_content-header-days\\">
  <li class=\\"font-medium\\">Mo</li>
  <li class=\\"font-medium\\">Tu</li>
  <li class=\\"font-medium\\">We</li>
  <li class=\\"font-medium\\">Th</li>
  <li class=\\"font-medium\\">Fr</li>
  <li class=\\"font-medium\\">Sa</li>
  <li class=\\"font-medium\\">Su</li>
</ul>"
`;

exports[`CalendarDays > renders calendarHeader properly 1`] = `
"<ul class=\\"calendar_wrapper_content-header-days\\">
  <li class=\\"font-medium\\">Mo</li>
  <li class=\\"font-medium\\">Tu</li>
  <li class=\\"font-medium\\">We</li>
  <li class=\\"font-medium\\">Th</li>
  <li class=\\"font-medium\\">Fr</li>
  <li class=\\"font-medium\\">Sa</li>
  <li class=\\"font-medium\\">Su</li>
</ul>"
`;
